tests/integration/test_design.py

```diff
@@ -3286,3 +3296,311 @@ def test_get_body_bounding_box(modeler: Modeler):
+        Interval.intersect(interval1, interval2, tolerance)
+    # Test intersection with negative intervals
+    interval1 = Interval(-10, -1)
+    interval2 = Interval(-12, -3)
+    intersection = Interval.intersect(interval1, interval2, tolerance)
+    assert intersection is not None
+    assert intersection.start == -10
+    assert intersection.end == -3
+
+    interval1.self_intersect(interval2, tolerance)
+    assert interval1.start == -10
+    assert interval1.end == -1
+
+
+def test_contains_value():
+    """Test the contains_value method of the Interval class."""
+    # Case 1: Value within the interval
+    interval = Interval(0, 10)
+    assert interval.contains_value(5, 0.1) is True
+    # Case 2: Value outside the interval
+    assert interval.contains_value(15, 0.1) is False
+    # Case 3: Value near the start boundary within accuracy
+    assert interval.contains_value(0.05, 0.1) is True
+    # Case 4: Value near the end boundary within accuracy
+    assert interval.contains_value(9.95, 0.1) is True
+    # Case 5: Value near the start boundary outside accuracy
+    assert interval.contains_value(-0.2, 0.1) is False
+    # Case 6: Value near the end boundary outside accuracy
+    assert interval.contains_value(10.2, 0.1) is False
+    # Case 8: Reversed interval (start > end)
+    reversed_interval = Interval(0, 10)  # Simulate reversed interval
+    reversed_interval._start = 10  # Manually override to simulate invalid state
+    reversed_interval._end = 0
+    assert reversed_interval.contains_value(5, 0.1) is True
+    # Case 9: Value exactly at the start
+    assert interval.contains_value(0, 0.1) is True
+    # Case 10: Value exactly at the end
+    assert interval.contains_value(10, 0.1) is True
+    # Case 11: Value less than start with tolerance
+    assert reversed_interval.contains_value(-2, 1) is False
+    # Case 12: Value greater than end with tolerance
+    assert reversed_interval.contains_value(11, 0.5) is False
+
+
+def test_param_uv_iter():
+    """Test the __iter__ method of ParamUV."""
+    param = ParamUV(3.5, 7.2)
+    u, v = param  # Unpack using the __iter__ method
+    assert u == 3.5
+    assert v == 7.2
+
+
+def test_param_uv_repr():
+    """Test the __repr__ method of ParamUV."""
+    param = ParamUV(3.5, 7.2)
+    assert repr(param) == "ParamUV(u=3.5, v=7.2)"
+
+
+def test_parameterization_repr():
+    """Test the __repr__ method of the Parameterization class."""
+    # Create a sample Parameterization object
+    interval = Interval(0, 10)
+    parameterization = Parameterization(ParamForm.CLOSED, ParamType.LINEAR, interval)
+    # Expected string representation
+    expected_repr = (
+        "Parameterization(form=ParamForm.CLOSED, type=ParamType.LINEAR, "
+        "interval=Interval(start=0, end=10))"
+    )
+    # Assert the __repr__ output matches the expected string
+    assert repr(parameterization) == expected_repr
+
+
+def test_planar_surface(modeler: Modeler):
```

Comment:
Remove modeler and move to test_parameterization.py (or test_shapes.py if you create it)